torchtext/experimental/transforms.py

```diff
@@ -422,3 +422,59 @@ def to_ivalue(self):
             vector = self.vector.to_ivalue()
             return VectorTransform(vector)
         return self
+
+
+class Pad(nn.Module):
+    r"""Pad sequences. Add padding id to the end of a sequences such that the sequences will
+        have the same length.
+
+    Args:
+        pad_id: the id for padding token
+        return_key_padding_mask: flag if to output padding_mask as output (Default: True)
+
+    Example:
+        >>> pad_id = 2
+        >>> pad_transform = Pad(pad_id)
+        >>> tensor_list = [torch.tensor([5, 4, 5, 6, 7]), torch.tensor([1, 3]), torch.tensor([7, 5, 8])]
+        >>> pad_seq, padding_mask = pad_transform(tensor_list)
+        >>> jit_pad_transform = torch.jit.script(pad_transform)
+    """
+
+    def __init__(self, pad_id, return_key_padding_mask=True):
```

Comment:
I'd make a choice here. Either return a data Tensor with padding values or return a data Tensor plus a mask Tensor.

The latter choice allows you to write code such

```
data, mask = Pad(0, True)
data.masked_fill(~mask, pad_id)
```

which means in essence specifying a padding is an optimization, but not a necessity.